tests/foreman/cli/test_errata.py

```diff
@@ -80,113 +74,107 @@
-                    'organization-id': self.org['id'],
-                    'host-ids': host['id'],
-                }
-            )
-        ActivationKey.add_host_collection(
+@pytest.fixture(scope='module')
+def setUp(module_org, module_activation_key):
+    """Create and setup host collection, hosts and virtual machines."""
+    virtual_machines = []
+    VIRTUAL_MACHINES_COUNT = 2
+    host_collection = make_host_collection({'organization-id': module_org.id})
+    for _ in range(VIRTUAL_MACHINES_COUNT):
+        # create VM
+        virtual_machine = VirtualMachine(distro=DISTRO_RHEL7)
+        virtual_machine.create()
+        virtual_machines.append(virtual_machine)
+        virtual_machine.install_katello_ca()
+        # register content host
+        virtual_machine.register_contenthost(
+            module_org['name'], module_activation_key=module_activation_key['name']
+        )
+        # enable red hat satellite repository
+        virtual_machine.enable_repo(REPOS['rhst7']['id'])
+        # install katello-agent
+        virtual_machine.install_katello_agent()
+        host = Host.info({'name': virtual_machine.hostname})
+        HostCollection.add_host(
             {
-                'id': self.activation_key['id'],
-                'host-collection-id': self.host_collection['id'],
-                'organization-id': self.org['id'],
+                'id': host_collection['id'],
+                'organization-id': module_org.id,
+                'host-ids': host['id'],
             }
         )
-        # install the custom package for each host
-        for virtual_machine in self.virtual_machines:
-            virtual_machine.run(f'yum install -y {self.CUSTOM_PACKAGE}')
-            result = virtual_machine.run(f'rpm -q {self.CUSTOM_PACKAGE}')
-            self.assertEqual(result.return_code, 0)
-
-    def _is_errata_package_installed(self, virtual_machine):
+    ActivationKey.add_host_collection(
+        {
+            'id': module_activation_key['id'],
+            'host-collection-id': host_collection['id'],
+            'organization-id': module_org.id,
+        }
+    )
+    # install the custom package for each host
+    for virtual_machine in virtual_machines:
+        virtual_machine.run(f'yum install -y {CUSTOM_PACKAGE}')
+        result = virtual_machine.run(f'rpm -q {CUSTOM_PACKAGE}')
+        assert result.return_code == 0
+
+    def _is_errata_package_installed(virtual_machine):
         """Check whether errata package is installed.
 
         :type virtual_machine: robottelo.vm.VirtualMachine
         :rtype: bool
         """
-        result = virtual_machine.run(f'rpm -q {self.CUSTOM_PACKAGE_ERRATA_APPLIED}')
+        result = virtual_machine.run(f'rpm -q {CUSTOM_PACKAGE_ERRATA_APPLIED}')
         return True if result.return_code == 0 else False
 
     @pytest.mark.tier3
-    def test_positive_install_by_hc_id_and_org_id(self):
+    def test_positive_install_by_hc_id_and_org_id(module_org, setup):
```

Comment:
Even if `self` isn't used within the test function, it is a good practice for every test function defined within a pytest class to take `self` as the first argument, before fixtures.

Reply:
Hi

I have removed the classes
I have reworked the fixtures, please review again